dbt/models/reporting/reporting.vw_pin_value_long.sql

```diff
@@ -1,76 +1,80 @@
 -- in long format
 
--- CCAO mailed, CCAO final, and BOR final values for each PIN by year.
--- We use ARBITRARY functions here to deduplicate PINs with multiple rows for
--- a given stage/pin/year combination. Values are always the same within these
--- duplicates.
+-- List of types of columns that we will extract when pivoting vw_pin_value
+-- from wide to long
+{% set cols = [
+    "class", "bldg", "land", "tot", "bldg_mv", "land_mv", "tot_mv"
+] %}
+
+-- Pivot vw_pin_value from wide to long, so that we create one row for each
+-- assessment stage with values pulled from the column types we defined above
 WITH stage_values AS (
     SELECT
-        parid,
-        taxyr,
-        REGEXP_REPLACE(class, '[^[:alnum:]]', '') AS class,
-        procname,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN ovrvalasm2
-                WHEN taxyr >= '2020' THEN valasm2
-            END
-        ) AS bldg,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN ovrvalasm1
-                WHEN taxyr >= '2020' THEN valasm1
-            END
-        ) AS land,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN ovrvalasm3
-                WHEN taxyr >= '2020' THEN valasm3
-            END
-        ) AS tot,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN NULL
-                WHEN taxyr >= '2020' THEN valapr2
-            END
-        ) AS bldg_mv,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN NULL
-                WHEN taxyr >= '2020' THEN valapr1
-            END
-        ) AS land_mv,
-        ARBITRARY(
-            CASE
-                WHEN taxyr < '2020' THEN NULL
-                WHEN taxyr >= '2020' THEN valapr3
-            END
-        ) AS tot_mv
-    FROM {{ source('iasworld', 'asmt_all') }}
-    WHERE procname IN ('CCAOVALUE', 'CCAOFINAL', 'BORVALUE')
-        AND rolltype != 'RR'
-        AND deactivat IS NULL
-        AND valclass IS NULL
-    GROUP BY parid, taxyr, procname, REGEXP_REPLACE(class, '[^[:alnum:]]', '')
+        t1.pin,
+        t1.year,
+        t2.class,
+        t2.stage_name,
+        t2.stage_num,
+        t2.bldg,
+        t2.land,
+        t2.tot,
+        t2.bldg_mv,
+        t2.land_mv,
+        t2.tot_mv
+    FROM {{ ref("default.vw_pin_value") }} AS t1
+    CROSS JOIN
+        UNNEST(
+            ARRAY[
+                'PRE-MAILED',
+                'MAILED',
+                'ASSESSOR PRE-CERTIFIED',
+                'ASSESSOR CERTIFIED',
+                -- This is a slightly different stage name from the name that
+                -- we use in vw_pin_value ('BOARD CERTIFIED'). At some point
+                -- we may want to align these names, but for now we maintain
+                -- the difference for legacy compatibility
+                'BOR CERTIFIED'
+            ],
+            ARRAY[0, 1, 1.5, 2, 3],
```

Comment:
i wonder if there's a way to pull these stage number values form `vw_pin_value`, but I'm sure you already tried.